### PR TITLE
Issue#SC-1474: Stronger rules for passwords

### PR DESCRIPTION
**Pull request by ArjunKhetia-NIIT**
Issue#SC-1474: Stronger rules for passwords

### CODE REVIEW

File: src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.html [hunk 1]
```diff
@@ -49,7 +49,8 @@
                 <i class=" icon">+91-</i>
               </div>
             </div>
-            <label class="ui basic blue error label pt-0" *ngIf="signUpForm.controls.password.touched && (passwordError != '')">{{passwordError}}</label>
+            <label class="ui basic blue error label pt-0 infotext" style="color: #007bff!important;" *ngIf="passwordError === ''">{{resourceService.frmelmnts.lbl.passwd}}</label>
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.html]: please remove the inline css

File: src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.scss [hunk 2]
```diff
@@ -210,4 +210,7 @@ input[type=number]::-webkit-inner-spin-button {
         font-weight: normal;
         cursor: pointer;
       }
+  }
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.scss]: please remove the css changes as we will not be using rem for calculating the size

If need please create a task for the UI Team and get their help to fix the UI

File: src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.ts [hunk 3]
```diff
@@ -150,17 +150,9 @@ export class SignupComponent implements OnInit, OnDestroy, AfterViewInit {
...
-        this.passwordError = this.resourceService.frmelmnts.lbl.passwd.lwcsError;
-      } else if (!upcsRegex.test(val)) {
-        this.passwordError = this.resourceService.frmelmnts.lbl.passwd.upcsError;
-      } else if (!numRegex.test(val)) {
-        this.passwordError = this.resourceService.frmelmnts.lbl.passwd.numError;
-      } else if (specRegex.test(val)) {
-        this.passwordError = this.resourceService.frmelmnts.lbl.passwd.specError;
+      const specRegex = new RegExp('^[^<>{}\'\"/|;:.\ ,~!?@#$%^=&*\\]\\\\()\\[¿§«»ω⊙¤°℃℉€¥£¢¡®©_+]*$');
+      if (!charRegex.test(val) || !lwcsRegex.test(val) || !upcsRegex.test(val) || !numRegex.test(val) || specRegex.test(val)) {
+        this.passwordError = this.resourceService.frmelmnts.lbl.passwd;
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.ts]: please use the _.get to get the Json opject

File: src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.html (hunk 1 above)
Comment (ArjunKhetia-NIIT) [reply src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.html]: Done

File: src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.scss (hunk 2 above)
Comment (ArjunKhetia-NIIT) [reply src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.scss]: Done

File: src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.ts (hunk 3 above)
Comment (ArjunKhetia-NIIT) [reply src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.ts]: Done

**Pull request closed without merging**